src/compile/legend.ts

```diff
@@ -162,15 +163,26 @@ export namespace properties {
       // For non-color legend, apply color config if there is no fill / stroke config.
       // (For color, do not override scale specified!)
       symbols[filled ? 'fill' : 'stroke'] = symbols[filled ? 'fill' : 'stroke'] ||
-        {value: model.config().mark.color};
+        {value: cfg.mark.color};
     }
 
     if (legend.symbolColor !== undefined) {
       symbols.fill = {value: legend.symbolColor};
+    } else if (symbols.fill === undefined) {
+      // fall back to mark config colors for legend fill
+      if (cfg.mark.color !== undefined) {
+        symbols.fill = {value: cfg.mark.color};
+      } else if (cfg.mark.fill !== undefined) {
+        symbols.fill = {value: cfg.mark.fill};
+      } else if (cfg.mark.stroke !== undefined) {
+        symbols.fill = {value: cfg.mark.stroke};
```

Comment:
Apologies for delayed reply, I think if the marks are hollow, the legend should consistently be hollow too.  Thank you.